services/src/main/java/org/keycloak/services/clientpolicy/executor/AutomaticClientRegistrationExecutor.java

```diff
@@ -0,0 +1,371 @@
+package org.keycloak.services.clientpolicy.executor;
+
+
+import java.util.List;
+
+import org.jboss.logging.Logger;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.keycloak.common.util.Base64Url;
+import org.keycloak.constants.AdapterConstants;
+import org.keycloak.events.EventBuilder;
+import org.keycloak.models.Constants;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.OAuth2Constants;
+import org.keycloak.OAuthErrorException;
+import org.keycloak.protocol.oidc.OIDCLoginProtocol;
+import org.keycloak.protocol.oidc.grants.ciba.CibaGrantType;
+import org.keycloak.representations.idm.ClientPolicyExecutorConfigurationRepresentation;
+import org.keycloak.representations.openid_federation.EntityStatement;
+import org.keycloak.representations.openid_federation.RPMetadata;
+import org.keycloak.services.clientpolicy.ClientPolicyContext;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+import org.keycloak.services.clientpolicy.context.PreAuthorizationRequestContext;
+import org.keycloak.services.clientregistration.AbstractClientRegistrationProvider;
+import org.keycloak.services.clientregistration.ClientRegistrationAuth;
+import org.keycloak.util.JsonSerialization;
+
+public class AutomaticClientRegistrationExecutor extends AbstractClientRegistrationProvider 
+                                implements ClientPolicyExecutorProvider<ClientPolicyExecutorConfigurationRepresentation> {
+
+    private static final Logger logger = Logger.getLogger(AutomaticClientRegistrationExecutor.class);
+
+    public AutomaticClientRegistrationExecutor(KeycloakSession session) {
+        super(session);
+        EventBuilder event = new EventBuilder(session.getContext().getRealm(), session);
+        super.setEvent(event);
+        super.setAuth(new ClientRegistrationAuth(session,this,event,"openid-connect"));
+    }
+
+    @Override
+    public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyException {
+
+        PreAuthorizationRequestContext paContext = null;
+        switch (context.getEvent()) {
+            case PRE_AUTHORIZATION_REQUEST:
+                paContext = (PreAuthorizationRequestContext) context;
+                String requestParam = paContext.getRequestParameters().getFirst(OIDCLoginProtocol.REQUEST_PARAM);
```

Comment:
Why not adding support also for _request_uri_ request parameter which is almost same as _request_ request parameter?